includes/discovery/sensors/fanspeed/benu.inc.php

```diff
@@ -0,0 +1,14 @@
+<?php
+
+// Adapated from Bluecoat sgos discovery
+
+echo 'Benu Sensors ';
+$sensor_index = 0;
+for ($index = 4; $index <= 9; $index++) { //Benu Fans are index 4 thru 9
+    $descr_oid   = "benuSensorName.1.$index";
+    $fan_oid     = ".1.3.6.1.4.1.39406.1.1.1.4.1.1.5.1.$index";
+    $descr       = snmp_get($device, $descr_oid, '-Oqv', 'BENU-CHASSIS-MIB');
+    $current     = snmp_get($device, $fan_oid, '-Oqv', 'BENU-CHASSIS-MIB');
```

Comment:
The snmpget test you've done above for benuSensorValue isn't using the MIB, you need to specify that with -m BENU-CHASSIS-MIB.

Yes you are correct that you will have two declarations of the OID.

The input for snmpget_multi isn't an array, it's just a space separated string for the OIDs. The output is an array though.